src/util/algorithms.cpp

```diff
@@ -5,6 +5,8 @@
 
 bool goto_functions_algorithm::run(goto_functionst &goto_functions)
 {
+  _goto_functions = goto_functions;
+  runOnProgram(goto_functions);
```

Comment:
This copy seems costly. Why is it necessary / where is it used?

Reply:
Hm, at the time it was because the Value-Set Analysis is a dependency for GCSE. I will invert the dependency though.